### PR TITLE
Elastic: Fixes so templating queries work

### LINKED ISSUE #30002: Elastic: Templating not working
**What happened**:
I was trying to migrate the Elastic datasource to use the new `fetch` api from `BackendSrv` and discovered that #28033 had introduced a bug in the `metricFindQuery` function as `parsedQuery` isn't used in the calls to `getFields` & `getTerms.` 
https://github.com/grafana/grafana/blob/375e8e4fd051a5c821c2b477634c73166d0694ab/public/app/plugins/datasource/elasticsearch/datasource.ts#L710
https://github.com/grafana/grafana/blob/375e8e4fd051a5c821c2b477634c73166d0694ab/public/app/plugins/datasource/elasticsearch/datasource.ts#L716 

Also the internal behavior of `getFields` has changed so now we need to pass in a string instead of an object.
https://github.com/grafana/grafana/compare/bb45f5f%5E..bb45f5f#diff-dcf3f853aa17281e6b8546ebed4c2ff0331ddc0afc8c98428091cd39932756cbR524

This was worsened by #29770 when using `DefaultTimeRange` that doesn't have valid `from` & `to` datetimes.
https://github.com/grafana/grafana/blob/375e8e4fd051a5c821c2b477634c73166d0694ab/public/app/plugins/datasource/elasticsearch/datasource.ts#L675 

**What you expected to happen**:
I would expect that templating would work.

**How to reproduce it (as minimally and precisely as possible)**:
1. start Elastic7 with devenv block: `make devenv sources=elastic7`
2. Create a new dashboard
3. Add a query variable with the following query `{"find": "terms", "field": "@hostname"}`
4. Expect to see the following: 
image
5. Result is: 
image
6. Checking the response from Elastic says that the dates are wrong:
image

**Environment**:
- Grafana version: Grafana v7.4.0-pre (375e8e4fd0)
- Data source type & version: Elastic7 (devenv block)
- OS Grafana is installed on: MacOS Catalina 10.15.7
- User OS & Browser: Chrome
- Grafana plugins: N/A
- Others:

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
As described in #30002 this PR fixes some bugs introduced in the Elastic data source. This PR also removes `DefaultTimeRange` and introduces `getDefaultTimeRange` function instead, this way we always get a valid default TimeRange. 

Enjoy the review 🎉 

**Which issue(s) this PR fixes**:
Fixes #30002

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/elasticsearch/datasource.ts [hunk 1]
```diff
@@ -142,7 +142,7 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
    *
    * @param url the url to query the index on, for example `/_mapping`.
    */
-  private get(url: string, range = DefaultTimeRange) {
+  private get(url: string, range = getDefaultTimeRange()) {
```
Comment (hugohaggmark) [inline public/app/plugins/datasource/elasticsearch/datasource.ts:145]: Part of the bug fix.

File: public/app/plugins/datasource/elasticsearch/datasource.ts [hunk 2]
```diff
@@ -667,7 +667,7 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
     });
   }
 
-  getTerms(queryDef: any, range = DefaultTimeRange) {
+  getTerms(queryDef: any, range = getDefaultTimeRange()) {
```
Comment (hugohaggmark) [inline public/app/plugins/datasource/elasticsearch/datasource.ts:670]: Part of the bug fix.

File: public/app/plugins/datasource/elasticsearch/datasource.ts [hunk 3]
```diff
@@ -707,13 +707,13 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
     if (query) {
       if (parsedQuery.find === 'fields') {
         parsedQuery.field = this.templateSrv.replace(parsedQuery.field, {}, 'lucene');
-        return this.getFields(query, range);
+        return this.getFields(parsedQuery, range);
```
Comment (hugohaggmark) [inline public/app/plugins/datasource/elasticsearch/datasource.ts:710]: Part of the bug fix.

File: public/app/plugins/datasource/elasticsearch/datasource.ts [hunk 4]
```diff
@@ -707,13 +707,13 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
...
       }
 
       if (parsedQuery.find === 'terms') {
         parsedQuery.field = this.templateSrv.replace(parsedQuery.field, {}, 'lucene');
         parsedQuery.query = this.templateSrv.replace(parsedQuery.query || '*', {}, 'lucene');
-        return this.getTerms(query, range);
+        return this.getTerms(parsedQuery, range);
```
Comment (hugohaggmark) [inline public/app/plugins/datasource/elasticsearch/datasource.ts:716]: Part of the bug fix.

**Review by Elfo404 — APPROVED** [review]
LGTM, thanks for catching this!

**Pull request merged**